backend/app/controllers/api/v1/courses_controller.rb

```diff
@@ -5,46 +5,52 @@ module V1
     class CoursesController < ApplicationController
       include Dry::Monads::Result::Mixin
 
+      skip_before_action :authenticate!, only: %i[all]
+
       def index
-        @courses = search_courses(Course.includes(:user).limit(100))
-        render json: prepare_json_for_list(@courses)
+        case course_search
+        in Success(*courses_found)
+          render json: courses_found, status: :ok
```

Comment:
courses_found is an array and using * we pass each element of the array as a separate argument. Then render each course?
E.g. courses_found = [course1, course2, course3]. Success(*courses_found) will render Sucess three times? Or I am wrong?

Reply:
If I clearly understand `*` in `DryMonads` use to capture an array value. You can read more about this here: https://dry-rb.org/gems/dry-monads/1.3/pattern-matching/